File: plugin-azure-agent-base/src/main/java/jetbrains/buildServer/clouds/azure/AzureUtils.java
package jetbrains.buildServer.clouds.azure;

import java.io.FileNotFoundException;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class AzureUtils {

    private static Pattern FILE_NOT_FOUND_CAUSE = Pattern.compile("\\(([^)]+)\\)");

    public static String getFileNotFoundMessage(FileNotFoundException e) {
        final String message = e.getMessage();
        final Matcher matcher = FILE_NOT_FOUND_CAUSE.matcher(message);
        if (matcher.find()) {
            return matcher.group(1);
        }

        return message;
    }
}

File: plugin-azure-agent/src/main/kotlin/jetbrains/buildServer/clouds/azure/AzureUtils.kt


package jetbrains.buildServer.clouds.azure

import java.io.FileNotFoundException

object AzureUtils {
    private val FILE_NOT_FOUND_CAUSE = Regex("\\(([^)]+)\\)")

    fun getFileNotFoundMessage(e: FileNotFoundException): String {
        val errorMessage = e.message!!
        FILE_NOT_FOUND_CAUSE.matchEntire(errorMessage)?.let {
            val (message) = it.destructured
            return message
        }

        return errorMessage
    }
}
